httpcore5-testing/src/test/java/org/apache/hc/core5/testing/compatibility/ContainerImages.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * ====================================================================
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * ====================================================================
+ *
+ * This software consists of voluntary contributions made by many
+ * individuals on behalf of the Apache Software Foundation.  For more
+ * information on the Apache Software Foundation, please see
+ * <http://www.apache.org/>.
+ *
+ */
+package org.apache.hc.core5.testing.compatibility;
+
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+import org.testcontainers.utility.DockerImageName;
+
+public final class ContainerImages {
+
+    public static final String AAA = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa";
+    public static final String BBB = "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb";
+    public static final String CCC = "ccccccccccccccccccccccccccccccccccccccccccccccccc";
+    public static final String PUSHY = "I am being very pushy";
+
+    public static int HTTP_PORT = 80;
+
+    public static GenericContainer<?> httpBin() {
+        return new GenericContainer<>(DockerImageName.parse("kennethreitz/httpbin:latest"))
```

Comment:
I don't know what the philosophy is in this project but as a rule of thumb I would avoid using `latest` - an upstream change could occur and break the build here with no code changes having occurred in this repo. Wouldn't it be better to use a specific version of this container instead?

Reply:
@massdosage The intention is to test our latest code with the latest images of _stable_ branches. I suppose `httpd:2.4` and `nginx:1.22` represent exactly that. `HttpBin` is a bit of a special beast as it changes very infrequently if ever and when it does I actually would like to know about it immediately. What versions would you see as more reasonable?

Reply:
I agree with @ok2c . While we're not focused on validating upstream changes, using latest for HttpBin makes sense due to its stability and infrequent updates.

Reply:
OK, if if it's stable and updated infrequently then I suppose this is fine. I've just been bitten many times when a build starts failing with no code changes in the repo due to a "latest" version changing in a backwards-incompatible way. This will never happen if you lock to a specific version. Sounds like this is a calculated risk that you're willing to take in this way and maybe it's even intentional to test against the latest upstream versions and you'd like to know if this breaks the code here in any way.

Reply:
There are also security implications of this, you're basically trusting all future versions of the upstream project. I work for a cybersecurity company so maybe I'm just being paranoid but having an opening like this in the "supply chain" for a project is also risky for other reasons.

Reply:
@massdosage The image is question is just a locally deployed version of http://httpbin.org/. What image version would you suggest? I will happily follow your recommendation.

Reply:
I would go with whatever the most recent fixed version is at this point in time, however now that I look into it, this `kennethreitz` container doesn't have any fixed versions, only `latest` and `test` (https://hub.docker.com/r/kennethreitz/httpbin/tags) :( So you don't have any other options for this specific container. The version provided by `kong` does have proper versions (https://hub.docker.com/r/kong/httpbin/tags) with `0.2.0` being the most recent but that's from a fork of `httpbin` which does look like it's being more actively maintained.

Reply:
@massdosage I think it used `latest` in the first because there was no better option.

Reply:
Fair enough, doesn't look there are any better options here, so `latest` it is!